core/Updates/5.0.0-b1.php

```diff
@@ -16,44 +16,49 @@
 use Piwik\Updater\Migration\Db as DbAlias;
 use Piwik\Updater\Migration\Db\DropIndex;
 use Piwik\Updater\Migration\Db\Sql;
+use Piwik\Updater\Migration\Factory;
 use Piwik\Updates as PiwikUpdates;
 
 /**
  * Update for version 5.0.0-b1
  */
 class Updates_5_0_0_b1 extends PiwikUpdates
 {
+    /**
+     * @var Factory
+     */
+    private $migration;
     private $tableName;
     private $indexName;
 
-    public function __construct()
+    public function __construct(Factory $factory)
     {
+        $this->migration = $factory;
+
         $this->tableName = Common::prefixTable('log_visit');
         $this->indexName = 'index_idsite_idvisitor';
     }
 
     public function doUpdate(Updater $updater)
     {
         if ($this->requiresUpdatedLogVisitTableIndex()) {
-            $this->updateLogVisitTableIndex();
+            $updater->executeMigrations(__FILE__, $this->getLogVisitTableMigrations());
         }
     }
 
-    private function updateLogVisitTableIndex()
+    private function getLogVisitTableMigrations()
```

Comment:
You should use the methods names that are defined in the base class, so the updater is able to determine the updates before they are executed. So there should be a `getMigrations` method that returns the migrations that are required to execute (it should return nothing if `requiresUpdatedLogVisitTableIndex` return true though) and the `doUpdate` should execute the updates, if any are returned from `getMigrations`

Reply:
@sgiehl I've update this to make use of the `getMigrations` method in the base class.